src/components/learner-credit-management/AssignmentsTableRefreshAction.jsx

```diff
@@ -1,11 +1,37 @@
 import React from 'react';
 import { Button } from '@edx/paragon';
 import PropTypes from 'prop-types';
+import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
+import { connect } from 'react-redux';
+import EVENT_NAMES from '../../eventTracking';
+import { applyFiltersToOptions, applySortByToOptions } from './data/hooks/useBudgetContentAssignments';
 
-const AssignmentsTableRefreshAction = ({ tableInstance, refresh }) => {
+const AssignmentsTableRefreshAction = ({ enterpriseId, tableInstance, refresh }) => {
   const handleRefresh = () => {
     const { state: dataTableState } = tableInstance;
     refresh(dataTableState);
+
+    // Construct track event data with a identical syntax to useBudgetContentAssignments
+    const options = {
+      page: dataTableState.pageIndex + 1, // `DataTable` uses zeo-indexed array
+      pageSize: dataTableState.pageSize,
+    };
+    applyFiltersToOptions(dataTableState.filters, options);
+    applySortByToOptions(dataTableState.sortBy, options);
+    const trackEventMetadata = {
+      filters: {
+        learnerState: options?.learnerState || null,
+        search: options?.search || null,
+      },
```

Comment:
nit/curious: why is `options` handled as if it may be undefined/null here whereas below the `?` is not included?

Reply:
I went ahead and continued the behavior for the rest of the attributes, and mirrored the same definition in the trackEventMetadata object in `useBudgetContentAssignments`